vcd/resource_vcd_catalog_vapp_template.go

```diff
@@ -377,6 +377,11 @@ func findVAppTemplate(d *schema.ResourceData, vcdClient *VCDClient, origin strin
 			if err != nil {
 				return nil, err
 			}
+			// This checks that the vApp Template is synchronized in the catalog
+			_, err = vcdClient.QueryVAppTemplateById(vAppTemplate.VAppTemplate.ID)
+			if err != nil {
+				return nil, fmt.Errorf("the found vApp Template %s is not synchronized in the catalog", vAppTemplate.VAppTemplate.ID)
```

Comment:
I'll revisit this part, thanks for your inputs!

Reply:
About

> One way to detect this case and do the right thing is by fetching the catalog ID (which does not change) and check whether the vApp template ID has changed. Not sure we want to make it so complex here, but at least we should know that it can happen.

The idea is to get the vApp Template by Catalog ID + Name and compare Old ID == New ID?

Reply:
Yes. As I said, it is complicated. Not sure whether we want to introduce this check here.
Testing this behavior is particularly difficult to reproduce in Terraform infrastructure. You have this issue only when `vcd_subscribed_catalog.make_local_copy` is false and the catalog has synchronized the list of items, but not their contents.

CC: @Didainius

Reply:
Yes. The only thing we need to be careful about is avoiding a temporary vApp template ID to get into Terraform state, as it might be difficult to get a clean state after that.